src/main/java/org/passay/dictionary/AbstractFileWordList.java

```diff
@@ -354,33 +354,52 @@ void put(final int index, final long position)
       if (map.position() == map.capacity()) {
         // 12 = 1.5 * 8 since this is a long view of a byte buffer
         final long newSize = map.capacity() * 12L;
-        if (newSize > Integer.MAX_VALUE) {
-          throw new IllegalArgumentException("Cache limit exceeded. Try reducing cacheSize.");
-        }
-        final LongBuffer temp = ByteBuffer.allocateDirect((int) newSize).asLongBuffer();
-        map.rewind();
-        temp.put(map);
-        map = temp;
+        resize(newSize);
       }
       map.put(position);
     }
 
 
     /**
-     * Gets the byte offset into the backing file for the word at the given index.
+     * Gets the byte offset into the backing file for the word at the index that is less than or equal to the supplied
+     * index.
      *
      * @param  index  Word at index.
      *
      * @return  Nearest cache entry for given index.
      */
     Entry get(final int index)
     {
-      if (modulus == 0) {
+      if (modulus == 0 || index < modulus) {
```

Comment:
I assume this is the fix to the couple tests that were failing. I don't see how that would matter, but the proof of the pudding is in the eating I suppose.